File: python/questao2.py
def extract_date_components(date):
    if len(date) != 10 or date[2] != '/' or date[5] != '/':
        print("Formato inválido. Utilize o formato DD/MM/AAAA.")
        return None, None, None

    day_str = date[:2]
    month_str = date[3:5]
    year_str = date[6:]

    try:
        day = int(day_str)
        month = int(month_str)
        year = int(year_str)
    except ValueError:
        print("DD, MM e AAAA devem ser numéricos.")
        return None, None, None

    return day, month, year

def main():
    date = input("Digite a data no formato DD/MM/AAAA: ")

    day, month, year = extract_date_components(date)

    if day is not None and month is not None and year is not None:
        print(f"Dia: {day}, Mês: {month}, Ano: {year}")
    else:
        print("Data inválida.")

if __name__ == "__main__":
    main()

File: c/questao2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

void extractDateComponents(char* date, int* day, int* month, int* year) {
    if (strlen(date) != 10 || date[2] != '/' || date[5] != '/') {
        printf("Formato inválido. Utilize o formato DD/MM/AAAA.\n");
        return;
    }

    char dayStr[3], monthStr[3], yearStr[5];
    dayStr[0] = date[0];
    dayStr[1] = date[1];
    dayStr[2] = '\0';

    monthStr[0] = date[3];
    monthStr[1] = date[4];
    monthStr[2] = '\0';

    yearStr[0] = date[6];
    yearStr[1] = date[7];
    yearStr[2] = date[8];
    yearStr[3] = date[9];
    yearStr[4] = '\0';

    *day = atoi(dayStr);
    *month = atoi(monthStr);
    *year = atoi(yearStr);
}

int main() {
    char date[11];
    int day, month, year;

    printf("Digite a data no formato DD/MM/AAAA: ");
    scanf("%s", date);

    extractDateComponents(date, &day, &month, &year);

    if (day <= 0 || day > 31 || month <= 0 || month > 12 || year <= 0) {
        printf("Data inválida.\n");
    } else {
        printf("Dia: %d, Mês: %d, Ano: %d\n", day, month, year);
    }

    return 0;
}
